### PR TITLE
feat: Adding Github workflows

### PR DESCRIPTION
**Pull request by lucifercr07**
### Changes done:
1. Added Linter
2. Added PR semantic validator
3. Added auto-assign PR plugin
4. Fixed `flake8` issues

### CODE REVIEW

File: .github/workflows/pre-commit-actions.yml
Comment (Patterbear) [inline .github/workflows/pre-commit-actions.yml:11]: I'm not too familiar with github actions, but will this line affect Windows compatibility?

File: .gitignore
Comment (Patterbear) [inline .gitignore:159]: JetBrains advises here not including .idea in the GITIGNORE - I'm not sure why because we obviously aren't meant to commit our own .idea folders. I think @rothoma2 will know more on this.

File: .github/workflows/pre-commit-actions.yml
Comment (lucifercr07) [reply .github/workflows/pre-commit-actions.yml:11]: Shouldn't it's just base platform to run on which to test `flake8` guidelines.

File: ms_office_macro_bleach/bleach.py
Comment (Patterbear) [inline ms_office_macro_bleach/bleach.py:1]: Looks good

File: README.md
Comment (Patterbear) [inline README.md:8]: 👍

File: .gitignore
Comment (lucifercr07) [reply .gitignore:159]: Sure, please let me know can revert. I use PyCharm and it was unnecessarily always in staging area.

**Review by Patterbear — COMMENTED** [review]
Thanks for requesting a review.

Unfortunately, I'm not very experienced with actions, however, this does look good.

Left a few comments just asking some questions. I'll wait on the boss' approval before approving.

File: .github/workflows/pre-commit-actions.yml
Comment (Patterbear) [reply .github/workflows/pre-commit-actions.yml:11]: Perfect then. Thanks for answering the question.

File: .gitignore
Comment (Patterbear) [reply .gitignore:159]: I use it too, and for my own projects always gitignore it. Just better double check with Robert - he'll probably know why JetBrains discourage it.

Comment (rothoma2) [reply .gitignore:159]: I also dont know, but lets keep the .idea on the GITIGNORE. If you want to use a specific IDE, keep the file locally and dont put it on the repo.

**Review by rothoma2 — APPROVED** [review]
Looks good.

**Pull request merged**
